packages/api/src/store/webhook-table.ts

```diff
@@ -38,4 +39,18 @@ export default class WebhookTable extends Table<DBWebhook> {
     });
     return { data: webhooks, cursor: nextCursor };
   }
+
+  async updateStatus(id: string, status: DBWebhook["status"]) {
+    const res = await this.db.query(
+      `UPDATE ${
+        this.name
+      } SET data = jsonb_set(data, '{status}', case when data->'status' is null then '{}' else data->'status' end || '${JSON.stringify(
+        status.status
```

Comment:
Since this function only updates the `status`, I don't think it should receive a full webhook object and then use only the `status` field of it. WDYT of removing the `.status` here and changing the callers to only send the `status` payload, and not an object that contains a `status` field?

I am also not sure how `TypeScript` compiler is allowing this, since `DBWebhook["status"]` object has no `status` field. Isn't it complaining about this line here? I guess on the call-site it does not complain because the `status` schema allows `additionalProperties` (the default is `true`), so it won't complain if you set properties that actually aren't there. 
Ah! And by explaining that I also understood how it is not complaining about anything right now. It is basically allowing you to call `status.status` here because of the additional properties, and for the same reason it is allowing you to send an object that doesn't actually match the `status` schema as the arguments here. To avoid that, WDYT of setting `additionalProperties: false` in the schema so it avoids these issues?

Edit: Also realized that, since typescript is not checking anything, the object you are sending for the `status` is actually not following the schema you defined on `schema.json`. You are setting an `errorMessage` field instead of an `error`, as it is in the schema right now. Just in case, let's add `additionalProperties: false` to all objects there unless we do need to allow additional properties. That means to both the `status` and the `lastFailure` objects.

Reply:
Yeah sounds good, I removed the status and send that only in the payload. I had some kind of trouble working on this query as it was nesting the objects and I took this path, but I think now that everything works properly I can send only that. Thanks for pointing it out & also noticing about the errorMessage and the error inconsistency, fixed it